fastly/pool.go

```diff
@@ -0,0 +1,293 @@
+	TLSClientKey     string   `mapstructure:"tls_client_key"`
+	TLSClientCert    string   `mapstructure:"tls_client_cert"`
+	TLSSNIHostname   string   `mapstructure:"tls_sni_hostname"`
+	TLSCheckCert     bool     `mapstructure:"tls_check_cert"`
+	TLSCertHostname  string   `mapstructure:"tls_cert_hostname"`
+	MinTLSVersion    string   `mapstructure:"min_tls_version"`
+	MaxTLSVersion    string   `mapstructure:"max_tls_version"`
+	Healthcheck      string   `mapstructure:"healthcheck"`
+	Type             PoolType `mapstructure:"type"`
+	OverrideHost     string   `mapstructure:"override_host"`
+}
+
+// poolsByName is a sortable list of pools.
+type poolsByName []*Pool
+
+// Len, Swap, and Less implement the sortable interface.
+func (s poolsByName) Len() int      { return len(s) }
+func (s poolsByName) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s poolsByName) Less(i, j int) bool {
+	return s[i].Name < s[j].Name
+}
+
+// ListPoolsInput is used as input to the ListPools function.
+type ListPoolsInput struct {
+	// Service is the ID of the service (required).
+	Service string
+
+	// Version is the specific configuration version (required).
+	Version int
+}
+
+// ListPools lists all pools for a particular service and version.
+func (c *Client) ListPools(i *ListPoolsInput) ([]*Pool, error) {
+	if i.Service == "" {
+		return nil, ErrMissingService
+	}
+
+	if i.Version == 0 {
+		return nil, ErrMissingVersion
+	}
+
+	path := fmt.Sprintf("/service/%s/version/%d/pool", i.Service, i.Version)
+	resp, err := c.Get(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var ps []*Pool
+	if err := decodeJSON(&ps, resp.Body); err != nil {
+		return nil, err
+	}
+	sort.Stable(poolsByName(ps))
+	return ps, nil
+}
+
+// CreatePoolInput is used as input to the CreatePool function.
+type CreatePoolInput struct {
+	// Service is the ID of the service. Version is the specific configuration
+	// version. Both fields are required.
+	Service string
+	Version int
+
+	// Name is the name of the pool to create (required).
+	Name string `form:"name"`
+
+	// Optional fields.
+	Comment          *string      `form:"comment,omitempty"`
+	Shield           *string      `form:"shield,omitempty"`
+	RequestCondition *string      `form:"request_condition,omitempty"`
+	MaxConnDefault   *uint        `form:"max_conn_default,omitempty"`
+	ConnectTimeout   *uint        `form:"connect_timeout,omitempty"`
+	FirstByteTimeout *uint        `form:"first_byte_timeout,omitempty"`
+	Quorum           *uint        `form:"quorum,omitempty"`
+	UseTLS           *Compatibool `form:"use_tls,omitempty"`
+	TLSCACert        *string      `form:"tls_ca_cert,omitempty"`
+	TLSCiphers       *string      `form:"tls_ciphers,omitempty"`
+	TLSClientKey     *string      `form:"tls_client_key,omitempty"`
+	TLSClientCert    *string      `form:"tls_client_cert,omitempty"`
+	TLSSNIHostname   *string      `form:"tls_sni_hostname,omitempty"`
+	TLSCheckCert     *bool        `form:"tls_check_cert,omitempty"`
```

Comment:
I think `TLSCheckCert` should be a `Compatibool` as well. The API does return a string `"0"/"1"` and not `false/true`.

Reply:
I think it is important that `TLSCheckCert` and `UseTLS` are of type `Compatibool` in the `Pool` struct declaration.

Reply:
I can confirm that `TLSCheckCert` returns "0/1" rather than "true/false", I'll update this field to use a `Compatibool`.

The [API docs](https://docs.fastly.com/api/dynamicservers#api-section-pool) note that the default is "0" for `UseTLS`, however the response style and default for `TLSCheckCert` isn't noted in the description.

Reply:
Fixed in 59ec1eb